### PR TITLE
Refactor secure randomness check

### LINKED ISSUE #5530: /dev/urandom warning seems unnecessary
Hello,

I've noticed there's a check for existence of `/dev/urandom` (pointed out by [this article](https://blog.housingsklave.at/?p=100)):

* https://github.com/nextcloud/server/blob/69b65042858eb8436a91af511c5e77cfeefff924/core/js/setupchecks.js#L107
* https://github.com/nextcloud/server/blob/e9b00f84b85e97283b83cf951f9bd262e627ae9a/settings/Controller/CheckSetupController.php#L147

It outputs *"/dev/urandom is not readable by PHP which is highly discouraged for security reasons. Further information can be found in our documentation"*.

However in PHP 7 on most recent platforms `/dev/urandom` is not used, and this check is confusing, especially in OpenBSD, as its web server is always chrooted, so there's no `/dev/urandom`. PHP 7 `random_bytes` and `random_int` use:

* `getrandom` on recent Linux
* `CryptGenRandom` on Windows
* `arc4random` on OpenBSD and NetBSD

On platforms that read randomness from `/dev/urandom` (like older Linux versions), the inability to open it by `random_bytes` will throw an exception: https://github.com/php/php-src/blob/696bd37e6757d77dc7ed44f3ea6451944ebaba96/ext/standard/random.c#L150

Since Nextcloud's SecureRandom uses `random_int`, it will throw if it fails to generate random bytes.

Proposal: instead of checking for existence of `/dev/urandom`, check if SecureRandom can successfully generate random bytes. This should be a setup error, not just a warning.

### PR DESCRIPTION
**Pull request by Hecke29**
closes #5530

### CODE REVIEW

**Comment by go2sh** [comment]
You don't need to update every translation file. It's done automatically.

**Comment by Hecke29** [comment]
@go2sh Do I have to execute some script to update them or just remove the commit entirely?

**Comment by go2sh** [comment]
There is a bot for the translations. You have to do nothing.

**Comment by Hecke29** [comment]
Removed the commit with those translations 👍

File: settings/Controller/CheckSetupController.php [hunk 1]
```diff
@@ -94,7 +97,8 @@ public function __construct($AppName,
 								EventDispatcherInterface $dispatcher,
 								IDBConnection $db,
 								ILockingProvider $lockingProvider,
-								IDateTimeFormatter $dateTimeFormatter) {
+								IDateTimeFormatter $dateTimeFormatter,
+								SecureRandom $secureRandom) {
```
Comment (nickvergessen) [inline settings/Controller/CheckSetupController.php]: Should use the interface `OCP\Security\ISecureRandom`

**Review by nickvergessen — CHANGES_REQUESTED** [review]

**Comment by Hecke29** [comment]
@nickvergessen done

**Pull request closed without merging**